### PR TITLE
Fix: nocheck for "fournisseur", already check with "societe"

### PR DESCRIPTION
**Pull request by hregis**
see issue https://github.com/Dolibarr/dolibarr/issues/8160

### CODE REVIEW

File: htdocs/core/lib/security.lib.php [hunk 1]
```diff
@@ -382,7 +382,7 @@ function checkUserAccessToObject($user, $featuresarray, $objectid=0, $tableandsh
 		$checkother = array('contact','agenda');	 // Test on entity and link to third party. Allowed if link is empty (Ex: contacts...).
 		$checkproject = array('projet','project'); // Test for project object
 		$checktask = array('projet_task');
-		$nocheck = array('barcode','stock');	// No test
+		$nocheck = array('barcode','stock','fournisseur');	// No test
```
Comment (eldy) [inline htdocs/core/lib/security.lib.php]: If you add this, test on permission for supplier orders or invoices are not done and everybody can see all record of all entities. For example in fourn/commande/list.php with the linen test return 
$result = restrictedArea($user, 'fournisseur', $orderid, '', 'commande');

Comment (hregis) [reply htdocs/core/lib/security.lib.php]: @eldy 
without "fournisseur" an error appears on the page "/product/fournisseurs.php" (onglet "prix d'achat") when the user is limited to seeing that the thirds which are attributed to him.

Param dbt_keyfield is required but not defined

if i add "dbt_keyfield" the user can not access to the page !

here "dbt_keyfield" is "fk_soc" (wrong solution)

$result=restrictedArea($user,'produit|service&fournisseur',$fieldvalue,'product&product','','fk_soc',$fieldtype);

**Comment by hregis** [comment]
@eldy 
it's ok with "fournisseur" in $checksoc,
it's ok for you ?

**Comment by hregis** [comment]
@eldy 
arf no ! same problem !

**Comment by hregis** [comment]
@eldy 
with this tab (product reseller price) you can check users "rights" for "fournisseur" but not the ID ! this is the ID of product !

**Comment by hregis** [comment]
@eldy 
it's ok with this PR

**Pull request merged**